feedsx/src/main/java/com/likeminds/feedsx/posttypes/util/PostTypeUtil.kt

```diff
@@ -566,24 +569,68 @@ object PostTypeUtil {
         }
     }
 
+    // setups the click listener on the link view
+    private fun setupLinkViewClick(
+        cvLinkPreview: MaterialCardView,
+        ivPlayYoutubeVideo: ImageView,
+        data: LinkOGTagsViewData
+    ) {
+        val context = cvLinkPreview.context
+        val url = data.url?.getUrlIfExist()
+        val videoId = url?.getYoutubeVideoId()
+
+        // shows the play button if link is of youtube and has a valid video id
+        if (!videoId.isNullOrEmpty()) {
+            ivPlayYoutubeVideo.show()
+        } else {
+            ivPlayYoutubeVideo.hide()
+        }
+
+        cvLinkPreview.setOnClickListener {
+            if (url?.isValidYoutubeLink() == true) {
+                // plays youtube video if link is of youtube and has a valid video id
+                if (!videoId.isNullOrEmpty()) {
+                    val extras = LMFeedYoutubePlayerExtras.Builder().videoId(videoId).build()
```

Comment:
shift `videoId()` and `build()` to next line, to maintain lint and code readability